### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by SalmanAsh**
This change is [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9)

### CODE REVIEW

File: src/pages/login/GitHub.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { Image } from "codeforlife/components"
+import { useLoginWithGitHubMutation } from "../../api/session"
+
+export interface GitHubProps {}
+
+const GitHub: FC<GitHubProps> = () => {
+  const [loginWithGitHub] = useLoginWithGitHubMutation()
+
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search)
```
Comment (SKairinos) [inline src/pages/login/GitHub.tsx:14]: not like this. we have a hook which helps you do this.

```js
import { useSearchParams } from "codeforlife/hooks/router"
import * as yup from "yup"
...
const searchParams = useSearchParams({
code: yup.string()
})
...
if (searchParams?.code) {
    // DO STUFF
}
```

File: src/pages/login/GitHub.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const code = urlParams.get("code")
+
+    if (code) {
+      urlParams.delete("code")
+      window.history.replaceState(null, "", window.location.pathname)
```
Comment (SKairinos) [inline src/pages/login/GitHub.tsx:19]: you must use the `useNavigate` hook to navigate between pages. you can only navigate after a component has mounted.

```js
import { useNavigate } from "codeforlife/hooks/router"
...
const navigate = useNavigate()
...
useEffect(() => {
// navigate to the same page with "."
navigate(".", { replace: true })
}, [...])
```

File: src/pages/login/GitHub.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+      loginWithGitHub({ code })
+        .unwrap()
+        .then(() => {
+          window.location.href = "/agreement-signatures"
```
Comment (SKairinos) [inline src/pages/login/GitHub.tsx:24]: navigate(paths.agreementSignatures._)

File: src/pages/login/GitHub.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        })
+        .catch(err => {
+          window.location.href = "/login"
```
Comment (SKairinos) [inline src/pages/login/GitHub.tsx:27]: You are already on the login page - there's no point in this.

File: src/pages/login/GitHub.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+          alert(`Login failed: ${err}`)
+        })
+    }
+  }, [loginWithGitHub])
+
+  return (
+    <pages.Page>
+      <pages.Section>
+        <div
```
Comment (SKairinos) [inline src/pages/login/GitHub.tsx:36]: Delete this div. Use MUI's [Stack](https://mui.com/material-ui/react-stack/) instead. You can add spacing between element's using the Stack's spacing property.

File: src/pages/login/GitHub.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+          style={{
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "center",
+            justifyContent: "center",
+            textAlign: "center",
+          }}
+        >
+          <Image src={CflLogoImage} alt="code for life logo" maxWidth="200px" />
+          <GitHubLoginButton />
```
Comment (SKairinos) [inline src/pages/login/GitHub.tsx:46]: Define button inline.

File: src/routes/login.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { Route } from "react-router-dom"
+
+import GitHub from "../pages/login/GitHub"
+import paths from "./paths"
+
+const login = (
+  <>
+    <Route path={paths.login._} element={<GitHub />} />
```
Comment (SKairinos) [inline src/routes/login.tsx:8]: <Route path={paths._} element={<Login />} />

File: src/routes/paths.ts [hunk 8]
```diff
@@ -1,9 +1,13 @@
...
 
 const paths = _("", {
-  fruits: _("/fruits", {
+  contributors: _("/contributors", {
     id: _("/:id"),
   }),
+  agreementSignatures: _("/agreement-signatures", {
+    id: _("/:id"),
+  }),
+  login: _("/login"),
```
Comment (SKairinos) [inline src/routes/paths.ts:10]: delete this.

File: src/pages/login/GitHub.tsx
Comment (SKairinos) [inline src/pages/login/GitHub.tsx:1]: delete this file and merge it's contents into the home page. the home will be our login page.

File: src/pages/home/Home.tsx [hunk 9]
```diff
@@ -19,7 +19,9 @@ const Home: FC<HomeProps> = () => {
           This is an example of how you can create a web page. This example
```
Comment (SKairinos) [inline src/pages/home/Home.tsx:19]: remove this sample text and add some brief text welcoming CFL contributor's our contributor service and encouraging them to become a contributor.

**Review by SKairinos — CHANGES_REQUESTED** [review]

**Review by SalmanAsh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9)* status: 0 of 28 files reviewed, 20 unresolved discussions (waiting on @SKairinos)

___
*[`src/pages/home/Home.tsx` line 19 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740923045:-O5n81_M6Rb3gewGRZ7V:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/home/Home.tsx#L19)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

remove this sample text and add some brief text welcoming CFL contributor's our contributor service and encouraging them to become a contributor.
</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 8 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740884671:-O5nGFaO16xMU_Dc5YO4:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L8)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

Do not need to be an independent, reusable component since it will only ever be rendered on the login page. Define this component inline in the login page and delete this file.
</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 13 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740870153:-O5n4ul-0moV2_Iw21In:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L13)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

sx = {theme => ({...})}
</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 16 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740882155:-O5nGCp9Eir8OnNKpJi-:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L16)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

don't use absolute values. use relative values.
```js
`${theme.spacing(4)} {theme.spacing(5)}`
```
[1 space = 8 pixels](https://github.com/ocadotechnology/codeforlife-package-javascript/blob/main/src/theme/spacing.ts)

</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 18 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740874814:-O5n54Cr85AcXywW6M3x:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L18)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

try "black" or "black.main". if that doesn't work, try theme.palette.black.main
</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 19 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740874415:-O5n54ws9ruAP6A3cIvg:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L19)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

try `"white"` or `"white.main"`. if that doesn't work, try `theme.palette.white.main`
</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 20 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740882095:-O5n56ry9OX1oDlKWwdW:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L20)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

why? doesn't seem necessary.
</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 21 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740882068:-O5n57PF4YxBUhb3Q3u-:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L21)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

why? doesn't seem necessary.
</blockquote></details>

Done.
___
*[`src/components/GitHubLoginButton.tsx` line 24 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740869649:-O5n5cXI9AEwFy9SawC-:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/components/GitHubLoginButton.tsx#L24)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

Use [this icon](https://mui.com/material-ui/material-icons/?query=github&selected=GitHub), not an image.
</blockquote></details>

Done.
___
*[`src/images/github-mark-white.png` line 0 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740885530:-O5n5hcO3QemSRSA_Bn-:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/images/github-mark-white.png#L0)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

delete this file. [use the icon that's already installed](https://mui.com/material-ui/material-icons/?query=github&selected=GitHub).
</blockquote></details>

Done.
___
*[`src/pages/login/GitHub.tsx` line 0 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740921277:-O5rN8R66upbEurNd_XH:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/login/GitHub.tsx#L0)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

delete this file and merge it's contents into the home page. the home will be our login page.
</blockquote></details>

Done.
___
*[`src/pages/login/GitHub.tsx` line 14 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740898033:-O5s-p1K-hpF4hddUEjl:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/login/GitHub.tsx#L14)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

not like this. we have a hook which helps you do this.

```js
import { useSearchParams } from "codeforlife/hooks/router"
import * as yup from "yup"
...
const searchParams = useSearchParams({
code: yup.string()
})
...
if (searchParams?.code) {
    // DO STUFF
}
```
</blockquote></details>

Done.
___
*[`src/pages/login/GitHub.tsx` line 19 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740905449:-O5rQ1ZF6QAg3knUnpSu:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/login/GitHub.tsx#L19)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

you must use the `useNavigate` hook to navigate between pages. you can only navigate after a component has mounted.

```js
import { useNavigate } from "codeforlife/hooks/router"
...
const navigate = useNavigate()
...
useEffect(() => {
// navigate to the same page with "."
navigate(".", { replace: true })
}, [...])
```
</blockquote></details>

Done.
___
*[`src/pages/login/GitHub.tsx` line 24 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740907173:-O5nG7I4EguA2KEL7Dyt:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/login/GitHub.tsx#L24)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

navigate(paths.agreementSignatures._)
</blockquote></details>

Done.
___
*[`src/pages/login/GitHub.tsx` line 27 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740908361:-O5nG6vgE_I7muZGaMMe:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/login/GitHub.tsx#L27)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

You are already on the login page - there's no point in this.
</blockquote></details>

Done.
___
*[`src/pages/login/GitHub.tsx` line 36 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740910572:-O5s1K0-2y-BldAWKuaX:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/login/GitHub.tsx#L36)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

Delete this div. Use MUI's [Stack](https://mui.com/material-ui/react-stack/) instead. You can add spacing between element's using the Stack's spacing property.
</blockquote></details>

Done.
___
*[`src/pages/login/GitHub.tsx` line 46 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740911006:-O5nG8e24YyCtT4_Nkew:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/pages/login/GitHub.tsx#L46)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

Define button inline.
</blockquote></details>

Done.
___
*[`src/routes/login.tsx` line 0 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740889135:-O5n8NlV901b6_xmUDC6:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/routes/login.tsx#L0)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

delete this file
</blockquote></details>

Done.
___
*[`src/routes/login.tsx` line 8 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740917919:-O5n8OGK9YVt92m_Aedj:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/routes/login.tsx#L8)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

`<Route path={paths._} element={<Home />} />`
</blockquote></details>

Done.
___
*[`src/routes/paths.ts` line 10 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#gh-1740918206:-O5n87Zs8IZr348bsWZ_:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/9074bf52e6f62ca9b490886841804d469a2ed75b/src/routes/paths.ts#L10)):*
<details><summary><i>Previously, SKairinos (Stefan Kairinos) wrote…</i></summary><blockquote>

delete this.
</blockquote></details>

Done.

**Review by SKairinos — COMMENTED** [review]
> Reviewed 13 of 23 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9)* status: 13 of 28 files reviewed, 1 unresolved discussion

**Review by SKairinos — CHANGES_REQUESTED** [review]
> Reviewed 5 of 27 files at r1, 10 of 23 files at r2.
> *[Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9)* status: all files reviewed, 10 unresolved discussions (waiting on @SalmanAsh)

___
*[`.vscode/launch.json` line 21 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O65zpFr3s1FRTpyClhj:-O65zpFs1_iNMKa7Dkrn:bvtpy32) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/.vscode/launch.json#L21)):*
> ```JSON
>     },
>     {
>       "env": {
> ```

delete this
___
*[`src/api/session.ts` line 10 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O66-18b296UbFhNzCAM:-O66-18b296UbFhNzCAN:bw75muy) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/api/session.ts#L10)):*
> ```XML
> const sessionApi = api.injectEndpoints({
>   endpoints: build => ({
>     loginWithGitHub: build.mutation<LoginWithGitHubResult, LoginWithGitHubArg>({
> ```

just login
___
*[`src/app/env.ts` line 5 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O6607vN18Uk0u5_Y6f3:-O6607vN18Uk0u5_Y6f4:br8zgxe) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/app/env.ts#L5)):*
> ```XML
> export * from "codeforlife/env"
> 
> export const GH_CLIENT_ID = env.GITHUB_CLIENT_ID ?? "REPLACE_ME"
> ```

- make the entire link an env var
- add the env var in the .env file
- variable names must me the same
- env vars must start with "VITE_"
___
*[`src/pages/home/Home.tsx` line 1 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O660mAr4l7IOygrEKBU:-O660mAr4l7IOygrEKBV:bckjp6m) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/pages/home/Home.tsx#L1)):*
> ```XML
> import * as pages from "codeforlife/components/page"
> ```

fix imports
___
*[`src/pages/home/Home.tsx` line 34 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O661D_X1fwrL6WMMXbq:-O661D_X1fwrL6WMMXbr:b-c00urr) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/pages/home/Home.tsx#L34)):*
> ```XML
>       navigate(paths.agreementSignatures._)
>     } else if (code) {
>       navigate(".", { replace: true })
> ```

move this into the catch below.
___
*[`src/pages/home/Home.tsx` line 53 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O661MQY6nf1IiGJ0p8P:-O661MQY6nf1IiGJ0p8Q:b-u7m6pa) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/pages/home/Home.tsx#L53)):*
> ```XML
>         <Stack
>           spacing={10}
>           sx={{
> ```

us stack-level props
___
*[`src/pages/home/Home.tsx` line 73 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O661TH2AthKBbIVvgP3:-O661TH2AthKBbIVvgP4:b-g6ejo7) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/pages/home/Home.tsx#L73)):*
> ```XML
>             to={`https://github.com/login/oauth/authorize?client_id=${GH_CLIENT_ID}`}
>             sx={theme => ({
>               marginTop: `${theme.spacing(20)}`,
> ```

no need for the string here
___
*[`src/pages/home/Home.tsx` line 74 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O661_OQ6oCf-L0cZMZ_:-O661_OQ6oCf-L0cZMZa:b-g6ejo7) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/pages/home/Home.tsx#L74)):*
> ```XML
>             sx={theme => ({
>               marginTop: `${theme.spacing(20)}`,
>               borderRadius: `${theme.spacing(1)}`,
> ```

no need for the string here
___
*[`src/pages/home/Home.tsx` line 76 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O661_q3-s8aUtEtqWa8:-O661_q3-s8aUtEtqWa9:b-g6ejo7) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/src/pages/home/Home.tsx#L76)):*
> ```XML
>               borderRadius: `${theme.spacing(1)}`,
>               padding: `${theme.spacing(4)} ${theme.spacing(5)}`,
>               fontSize: `${theme.spacing(2.5)}`,
> ```

no need for the string here
___
*[`.env` line 2 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9#-O661ooo3xC1iVyZIcRv:-O661ooo3xC1iVyZIcRw:b-13ro87) ([raw file](https://github.com/ocadotechnology/codeforlife-contributor-frontend/blob/35e75b9f2949cef59ba08b3e8c195b128fd08c72/.env#L2)):*
> ```Dotenv
> # Service.
> VITE_SERVICE_NAME=contributor
> ```

add VITE_LINK_GH_LOGIN

**Review by SKairinos — APPROVED** [review]
> Reviewed 16 of 16 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-contributor-frontend/9)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @SalmanAsh)

**Pull request merged**